react/features/calendar-sync/middleware.js

```diff
@@ -3,13 +3,12 @@
 import RNCalendarEvents from 'react-native-calendar-events';
 
 import { APP_WILL_MOUNT } from '../app';
-import { SET_ROOM } from '../base/conference';
+import { getKnownDomains } from '../base/domains';
```

Comment:
shouldn't this middleware handle the add known domains action, so if you join a server which is not on the list, we start displaying those events too? Or is this accomplished in a different way?

Reply:
The calendar list is refreshed (and entries fetched) every time you get to the welcome page. When a new domain is added, we're not on the welcome page, so when we get back, the new list will be rendered with the newly added domains in mind.

Reply:
Come to think of it, I don't think I like that inter dependency. We are using the calendar thing on the welcome page now, but as a feature, its data should be accurate whether we use the welcome page or not. If we ever want to add a sliding panel to the conference view it wouldn't take effect until the next conference, which is not very nice.

Reply:
I also thought it through, and I agree, but not because of a future feature, but because of the calendar notification feature. The scenario when we can fail if we don't add this action:
1. User joins a meeting on a new server
2. The user has an adjacent meeting in his/her calendar with the same server name, that he/she won't be alerted for, because we don't have an up-to-date calendar list.

So thanks for making me think of it, added the action!